tests/Integration/Behaviour/Features/Scenario/Product/Combination/Shop/delete_multishop_combination.feature

```diff
@@ -0,0 +1,281 @@
+    And attribute "White" named "White" in en language exists
+    And attribute "Black" named "Black" in en language exists
+    And attribute "Blue" named "Blue" in en language exists
+    And shop "shop1" with name "test_shop" exists
+    And multistore feature is enabled
+    And shop group "default_shop_group" with name "Default" exists
+    And I add a shop "shop2" with name "default_shop_group" and color "red" for the group "default_shop_group"
+    And I add a shop group "test_second_shop_group" with name "Test second shop group" and color "green"
+    And I add a shop "shop3" with name "test_third_shop" and color "blue" for the group "test_second_shop_group"
+    And I add a shop "shop4" with name "test_shop_without_url" and color "blue" for the group "test_second_shop_group"
+    And single shop context is loaded
+    Given I add product "product1" with following information:
+      | name[en-US] | universal T-shirt |
+      | type        | combinations      |
+    And product product1 type should be combinations
+    And I copy product product1 from shop shop1 to shop shop2
+    And I generate combinations in shop "shop1" for product product1 using following attributes:
+      | Size  | [S,M]              |
+      | Color | [White,Black,Blue] |
+    Then product "product1" should have the following combinations for shops "shop1":
+      | id reference   | combination name        | reference | attributes           | impact on price | quantity | is default |
+      | product1SWhite | Size - S, Color - White |           | [Size:S,Color:White] | 0               | 0        | true       |
+      | product1SBlack | Size - S, Color - Black |           | [Size:S,Color:Black] | 0               | 0        | false      |
+      | product1SBlue  | Size - S, Color - Blue  |           | [Size:S,Color:Blue]  | 0               | 0        | false      |
+      | product1MWhite | Size - M, Color - White |           | [Size:M,Color:White] | 0               | 0        | false      |
+      | product1MBlack | Size - M, Color - Black |           | [Size:M,Color:Black] | 0               | 0        | false      |
+      | product1MBlue  | Size - M, Color - Blue  |           | [Size:M,Color:Blue]  | 0               | 0        | false      |
+    And product "product1" default combination for shop "shop1" should be "product1SWhite"
+    And product "product1" should have no combinations for shops "shop2"
+    When I generate combinations in shop "shop2" for product product1 using following attributes:
+      | Size  | [S,M]              |
+      | Color | [White,Black,Blue] |
+    Then product "product1" should have the following combinations for shops "shop2":
+      | id reference        | combination name        | reference | attributes           | impact on price | quantity | is default |
+      | product1SWhiteShop2 | Size - S, Color - White |           | [Size:S,Color:White] | 0               | 0        | true       |
```

Comment:
```suggestion
      | product1SWhite | Size - S, Color - White |           | [Size:S,Color:White] | 0               | 0        | true       |
```

Reply:
Actually, why do we need a new reference here? The ID has already been stored with the previous reference `product1SWhite` and already contains the appropriate Combination ID. So if we use the same here we do validate that the ID didn't change actually no?

This would simplify the writing/reading of the scenarios AND validate something at the same time.

Reply:
Aaahh but the problem is that if the reference is not found we automatically use it as the storage reference.

Then I think we should improve this:
- `id reference` is used only when we want to really store the reference
- `combination id` is used to assert that the values match

This way you can use `id reference` on the first assert which saves combination IDs and the following ones only use `combination id` so it validates that the IDs remain the same